open_instruct/dataset_transformation.py

```diff
@@ -826,6 +838,79 @@ def sft_tulu_tokenize_and_truncate_v1(row: Dict[str, Any], tokenizer: PreTrained
     return row
 
 
+def sft_span_seach_mask_out(row: Dict[str, Any], tokenizer: PreTrainedTokenizer, max_seq_length: int):
+
```

Comment:
I would suggest to add a docstring adapting from that of the companion function `sft_tulu_tokenize_and_truncate_v1`.

>"""
    This function encodes a single example into a format that can be used for sft training.
    Here, we assume each example has a 'messages' field. Each message in it is a dict with 'role' and 'content' fields.
    We use the `apply_chat_template` function from the tokenizer to tokenize the messages and prepare the input and label tensors.
    """

A suggestion:

> This function encodes a single example into a format that can be used for sft training (similar to `sft_tulu_tokenize_and_truncate_v1`). Instead of performing label masking iteratively, this function performs masking via span search and can handle complex chat templates with thinking.